src/borg/manifest.py

```diff
@@ -36,23 +36,27 @@ class MandatoryFeatureUnsupported(Error):
 
 def filter_archives_by_date(archives, oldest=None, newest=None, older=None, newer=None):
     def get_first_and_last_archive_dates(archives_list):
-        dates = [x.ts.date() for x in archives_list]
+        dates = [x.ts for x in archives_list]
         return min(dates), max(dates)
 
-    today = archive_ts_now().date()
+    today = archive_ts_now()
```

Comment:
the names in this function still remind of being capable to compute only dates (like day granularity).

how about:

- today -> now (or now_ts, if you prefer)
- *_date -> *_ts